Tracker/tracker.lua

```diff
@@ -8567,6 +8653,25 @@ function tracker:updateNames()
   end
 end
 
+function tracker:getNoteNames()
+  -- track_num as returned here is one-indexed, but as used by GetTrackMIDINoteName, it's zero-indexed (christ)
+  local track_num = reaper.GetMediaTrackInfo_Value(tracker.track, 'IP_TRACKNUMBER') - 1
+
+  -- can't actually tell whether channel makes a difference here?
+  -- sorta seems like it does not, but whatever
+  local channel
+  if tracker.outChannel >= 1 then
+    channel = tracker.outChannel - 1
+  else
+    channel = 0
+  end

```

Comment:
I'm not 100% sure whether this does anything (since I can't find a way in reaper to set per channel note names), but just for future reference, when channel = 0, it means that the channel that we're emitting on is the note column index (not channel 0).

This channel is given by the second return of tracker:getLocation(). First check the fieldtype whether you're in a channel part though (fieldtypes are: text, vel1, vel2, delay1, delay2, end1, end2).

local ftype, chan, row = self:getLocation()

This comes with one difficulty, which is that it'd require note name fetching every time a user changes note column when channel is set to zero, which may incur some overhead.

Do you know how to set per channel note names?

Reply:
I don't know anything about setting per-channel note names. Looking at the API, SetTrackMIDINoteName takes a chan arg, so it seems that it's possible. Its documentation has this to say: "channel < 0 assigns these note names to all channels"; I guess that's what rs5k does. Per-channel note names seem like such a niche use case that I can't really be bothered to look into whether it works or not - though I'm happy to spend time on it if anyone should actually request it. In which case, I wonder if the channel-checking stuff here should just be taken out, and 0 should be used regardless of the tracker's channel?